### ISSUE TITLE
Strange behavior of class QuoteRequestBuilder

### ISSUE THREAD

**Issue by t13ka** [VirtoCommerce/vc-storefront-core#465]
I always thought that the builder should build something but on the contrary, we have different actions here which are not really correlated with the name. In my opinion, the builder should return some built object/s. This class serves as the storage or proxy or service or manager. It is quite hard to understand the idea

https://github.com/VirtoCommerce/vc-storefront-core/blob/84ce8d7393b8b01cd23822208a955624e0fe0078/VirtoCommerce.Storefront/Domain/Quote/QuoteRequestBuilder.cs#L24

I suggest try this out for future refactoring: https://refactoring.guru/design-patterns/builder/csharp/example

**Comment by mvktsk** [comment]
Task https://virtocommerce.atlassian.net/browse/VP-3569 has been created

**Comment by yecli** [comment]
I agree on this, looking at the interface methods. probably Manager is better.
But I doubt we could change it, as it would be breaking change.
@tatarincev What do you think, could we change this interface name?

**Comment by mvktsk** [comment]
Task https://virtocommerce.atlassian.net/browse/VP-3713 has been created
